patchdeps.py

```diff
@@ -149,7 +149,10 @@ def print_depends_tsort(patches: list[Changeset], depends: dict[Changeset, dict[
         if dependencies := depends[p]:
             for dep in patches:
                 if dep in dependencies:
-                    print(f"{dep}\t{p}")
+                    def no_delim(x):
```

Comment:
Good find; thank you for tackling this. Only some small nits:

1. Move the function definition out of the `for`-loop
2. Please also add type-hints here; `mypy` / `pytype` / … will complain otherwise.
3. Maybe better use `str.translate` instead of a chain of `str.replace()?
4. Rename the function to start with a verb?
5. Code-formaters like `ruff` will probably complain, if there is no empty line before and after the (nested) function.

```diff
+
+    def replace_tr_delim(cs: Changeset) -> str:
+        # Tsort source has: #define DELIM " \t\n"
+        return str(cs).translate(str.maketrans(" \t\n", "___"))
+
     for p in patches:
         if dependencies := depends[p]:
             for dep in patches:
                 if dep in dependencies:
-                    def no_delim(x):
-                        # Tsort source has: #define DELIM " \t\n"
-                        return str(x).replace(' ', '_').replace('\t', '_').replace('\n', '_')
-                     print(f"{no_delim(dep)}\t{no_delim(p)}")
+                     print(f"{replace_tr_delim(dep)}\t{replace_tr_delim(p)}")